docs/topics/classes.md

```diff
@@ -1,245 +1,692 @@
-        // draw the rectangle
-    }
-}
+Like any other class, abstract classes can also have constructors. 
+These constructors initialize class properties and enforce required parameters for subclasses:
+
+```kotlin
+abstract class Person(val name: String, val age: Int)
 ```
 
-You can override a non-abstract `open` member with an abstract one.
+An abstract class can have both abstract and non-abstract members (properties and functions). 
+To declare a member as abstract, you must use the `abstract` keyword explicitly.
+
+Abstract members do not have an implementation in the abstract class. The `override`
+keyword ensures that a subclass provides an actual implementation for an abstract function:
 
 ```kotlin
-open class Polygon {
-    open fun draw() {
-        // some default polygon drawing method
+// Abstract class with a primary constructor that defines 'name' and 'age'
+abstract class Person(
+    val name: String,
+    val age: Int
+) {
+    // Abstract member (doesn't provide default implementation, must be implemented by subclasses)
+    abstract fun introduce()
+
+    // Non-abstract member (has a default implementation)
+    fun greet() {
+        println("Hello, my name is $name.")
+        // Hello, my name is Alice.
+    }
+}
+
+// Subclass that provides an implementation for the abstract member
+class Student(
+    name: String,
+    age: Int,
+    val school: String
+) : Person(name, age) {
+    override fun introduce() {
+        println("I am $name, $age years old, and I study at $school.")
+        // I am Alice, 20 years old, and I study at Engineering University.
     }
 }
 
-abstract class WildShape : Polygon() {
-    // Classes that inherit WildShape need to provide their own
-    // draw method instead of using the default on Polygon
-    abstract override fun draw()
+fun main() {
+    // Creates an object of the Student class
+    val student = Student("Alice", 20, "Engineering University")
+    // Calls the non-abstract member
+    student.greet()
+    // Calls the overridden abstract member
+    student.introduce()
 }
 ```
+{kotlin-runnable="true"}
+
+You don't need to annotate abstract classes or functions with the `open` keyword because they are implicitly 
+open and designed to be inherited. 
+
+[For more details about the `open` keyword, see Inheritance](inheritance.md#open-keyword).
 
 ## Companion objects
 
-If you need to write a function that can be called without having a class instance but that needs access to the internals
-of a class (such as a factory method), you can write it as a member of an [object declaration](object-declarations.md) inside that class.
+In Kotlin, a [companion object](object-declarations.md#companion-objects) is a type of object declaration 
+that allows you to access its members using the class name without needing an object from the class.
+
+If you need to write a function that can be called without creating an object of a class but still needs access to the 
```

Comment:
I would rephrase this to smth like "If you need to write a function that can be called without creating an object of a class but is still logically connected to the class (such as a factory function), you can define ..."